packages/edit-post/src/components/sidebar/plugin-document-setting-panel/index.js

```diff
@@ -30,8 +30,9 @@ const PluginDocumentSettingFill = ( { isEnabled, panelName, opened, onToggle, cl
 						className={ className }
 						title={ title }
 						icon={ icon }
-						opened={ opened }
-						onToggle={ onToggle }
+						initialOpen={ initialOpen }
+						opened={ initialOpen ? undefined : opened }
```

Comment:
This feels a little weird for me -- why do we need to set it to `undefined` if the `initialOpen` prop exists?

Reply:
I did this because of how the interior component `PanelBody` is set up.

For the opened property:
https://github.com/WordPress/gutenberg/blob/b45e53e97fe5c22b3915fcc12ca740085d3a1141/packages/components/src/panel/body.js#L41-L42
If the opened prop is `undefined` it falls back to using its default behavior of relying on state as opposed to the store connection.  So if we did not set it to `undefined` in this declaration, it would overwrite the behavior of `initialOpen`.

Similarly, for the toggle function: 
https://github.com/WordPress/gutenberg/blob/b45e53e97fe5c22b3915fcc12ca740085d3a1141/packages/components/src/panel/body.js#L27-L38

If `opened` is `undefined` we rely on the components state as opposed to the store connection.  And `onToggle` will run if it is not `undefined`.

Essentially, having these properties defined will overwrite the behavior we want from `initialOpen`.  So if `initialOpen` is truthy, we need to explicitly set these as `undefined` so they don't overwrite its behavior.

Reply:
ah interesting. Cool!

Reply:
Unfortunately It looks like when the panel is using store state, the open/closed state is synced to local storage. This seems to circumvent that, which I don't think is a positive thing.